daemon/daemon_test.go

```diff
@@ -335,6 +337,54 @@ func TestDaemonReloadLabels(t *testing.T) {
 	}
 }
 
+func TestDaemonReloadInsecureRegistries(t *testing.T) {
+	daemon := &Daemon{}
+	daemon.RegistryService = registry.NewService(registry.ServiceOptions{})
+	daemon.configStore = &Config{}
```

Comment:
I polished the test case and add some comments there, PTAL @cpuguy83 
If it still has some places not so good, please tell me and I will try to polish it gain.